x-pack/plugins/reporting/common/types.d.ts

```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export { ConfigType } from '../server/config';
```

Comment:
A new file for sharing types in server / client. I expect there will be more additions to this in the future when we remove the code from legacy and clean up where types are defined